tests/unit/store/unload-test.js

```diff
@@ -97,6 +97,39 @@ test("unload a record", function(assert) {
   });
 });
 
+test("unload a record - observers should be removed", function(assert) {
+  assert.expect(3);
+
+  run(function() {
+    store.push({
+      data: {
+        type: 'record',
+        id: '1',
+        attributes: {
+          title: 'toto'
+        }
+      }
+    });
+
+    store.findRecord('record', 1).then(function(record) {
+      Ember.addObserver(record, 'title', record, function() {
+        assert.ok(false, 'observer for record.title');
+      });
```

Comment:
Can you add an assertion for `obersvers.length` being `1` here? Just to be sure that the assertion below does what we expect it to do...